src/lib.rs

```diff
@@ -58,14 +58,17 @@ where
             return Err(Error::WrongAddress);
         }
 
-        // Enable all axes, normal mode.
-        lis3dh.write_register(Register::CTRL1, 0x07)?;
-        // Set 400Hz data rate.
+        // Use normal mode.
+        lis3dh.set_mode(Mode::Normal)?;
+        // Set 400Hz output data rate.
         lis3dh.set_datarate(DataRate::Hz_400)?;
-        // High res & BDU enabled
-        lis3dh.write_register(Register::CTRL4, 0x88)?;
+        // Enable all axes.
+        lis3dh.enable_axis((true, true, true))?;
+
+        // Block data update & High-resolution output mode enabled
+        lis3dh.write_register(Register::CTRL4, BDU & HR)?;
```

Comment:
Why are we mucking with High res mode outside of set_mode?

Reply:
I'm not entirely sure, there was a bunch of initialization in the constructor and I had just left it in place. I think a lot of the settings are default values anyways, so I'm not sure if it's appropriate to have them in there or not. Might be better to supply a `reset` function or something of that nature.

Reply:
Yeah these definitely arent your changes. But they're definitely not default values.

Reply:
Just FYI, I've given you access to my fork if you're interested. I am looking into the issues, but feel free to do whatever.